hack/validate.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+set -eo pipefail
+
+eval $(go env)
+PATH="$GOPATH/bin:$PATH"
+
+die() { echo "Error: ${1:-No message provided}" > /dev/stderr; exit 1; }
+
+# Always run from the repository root
+cd $(dirname "${BASH_SOURCE[0]}")/../
+
+if [[ -z $(type -P gofmt) ]]; then
+    die "Unable to find 'gofmt' binary in \$PATH: $PATH"
+fi
+
+echo "Executing go vet"
+GO111MODULE="on" go vet ./...
+
+echo "Executing gofmt"
+OUTPUT=$(gofmt -s -l . |& sed -r -e '/^vendor/d')
```

Comment:
One last thing,
```suggestion
OUTPUT=$(gofmt -s -l . 2>&1 | sed -r -e '/^vendor/d')
```
please, macOS ships a very old `bash` (and probably isn’t going to upgrade), and this runs from `make` during ordinary development.

(Does the `-r` do anything here?)

Reply:
Oh, didn't realize that.  It's probably not necessary, same with the `-r`...

Reply:
...okay it's simplified now.  I don't have a Mac to test it on though.

Reply:
Tested this manually (success, invalid Gofmt flags, file not in the correct format), everything is as expected.